pkg/operator/kubeletversionskewcontroller/kubelet_version_skew_controller.go

```diff
@@ -0,0 +1,140 @@
+	if err != nil {
+		return err
+	}
+
+	var errorMsgs []string
+	var skewedToTheLimitMsgs []string
+	var skewedButOKMsgs []string
+	var syncedMsgs []string
+	var unsupportedMsgs []string
+
+	// for each node, check kubelet version
+	for _, node := range nodes {
+		kubeletVersion, err := nodeKubeletVersion(node)
+		skew := int(kubeletVersion.Minor - c.apiServerVersion.Minor)
+		switch {
+		case err != nil:
+			errorMsgs = append(errorMsgs, fmt.Sprintf("unable to determine kubelet version on node %s: %s", node.Name, err))
+		case skew <= -2: // behind too far
+			skewedToTheLimitMsgs = append(skewedToTheLimitMsgs, fmt.Sprintf("Kubelet version (%s) on node %s is behind the expected API server version (%s) by at least the maximum allowable skew.", kubeletVersion, node.Name, c.apiServerVersion))
+		case skew == -1: // behind, but ok
+			skewedButOKMsgs = append(skewedButOKMsgs, fmt.Sprintf("Kubelet version (%s) on node %s is behind the expected API server version (%s), but within the maximum allowable skew.", kubeletVersion, node.Name, c.apiServerVersion))
+		case skew == 0: // synced
+			syncedMsgs = append(syncedMsgs, fmt.Sprintf("Kublet version on node %s is in synce with the expected API server version.", node.Name))
+		default: // kubelet version newer than api server version. possibly a rollback.
+			unsupportedMsgs = append(unsupportedMsgs, fmt.Sprintf("Kubelet version (%s) on node %s is ahead of the expected API server version (%s).", kubeletVersion, node.Name, c.apiServerVersion))
+		}
+	}
+
+	// update status
+
+	// combine all msgs
+	msg := strings.Join(append(errorMsgs, append(skewedToTheLimitMsgs, append(skewedButOKMsgs, append(syncedMsgs, unsupportedMsgs...)...)...)...), "\n")
+	condition := operatorv1.OperatorCondition{Type: KubeletVersionSkewLimitUpgradeableConditionType, Message: msg}
+	// use the most "severe" reason to set the condition status
+	switch {
+	case len(errorMsgs) > 0:
+		condition.Reason = KubeletVersionUnknownReason
+		condition.Status = operatorv1.ConditionUnknown
+	case len(unsupportedMsgs) > 0:
+		condition.Reason = KubeletVersionUnsupportedReason
+		condition.Status = operatorv1.ConditionUnknown
+	case len(skewedToTheLimitMsgs) > 0:
+		condition.Reason = KubeletVersionSkewAtLimitReason
+		condition.Status = operatorv1.ConditionFalse
+	case len(skewedButOKMsgs) > 0:
+		condition.Reason = KubeletVersionSkewLessThanLimitReason
+		condition.Status = operatorv1.ConditionTrue
+	default:
+		condition.Reason = KubeletVersionSyncedReason
+		condition.Status = operatorv1.ConditionTrue
+		condition.Message = "Kubelet and API server versions are synced."
+	}
+
+	_, _, err = v1helpers.UpdateStatus(c.operatorClient, v1helpers.UpdateConditionFn(condition))
+	return err
+
+}
+
+func nodeKubeletVersion(node *corev1.Node) (semver.Version, error) {
+	return semver.Parse(strings.TrimPrefix(node.Status.NodeInfo.KubeletVersion, "v"))
+}
+
+func (c *kubeletVersionSkewController) calculateSkew(kubeletVersion semver.Version) uint64 {
```

Comment:
`calculateSkew` is not being used, it seems.

Reply:
Deleted.